### PR TITLE
add sslsocketfactory, hostnameverifier for self signed certificate su…

### PR DESCRIPTION
**Pull request by ninekaw9**
…pport

## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: sentry/src/main/java/io/sentry/transport/HttpTransport.java [hunk 1]
```diff
@@ -131,6 +146,8 @@ public HttpTransport(
     this.readTimeout = readTimeoutMillis;
     this.options = options;
     this.bypassSecurity = bypassSecurity;
+    this.sslSocketFactory = sslSocketFactory;
```
Comment (bruno-garcia) [inline sentry/src/main/java/io/sentry/transport/HttpTransport.java:149]: Do you mind also deleting all this `bypassSecurity` code?
You approach is preferable and we'll shift over to it.

**Review by bruno-garcia — COMMENTED** [review]
Would be nice to get some tests for this

**Comment by bruno-garcia** [comment]
This is a great addition, thanks!

File: sentry/src/main/java/io/sentry/transport/HttpTransport.java (hunk 1 above)
Comment (ninekaw9) [reply sentry/src/main/java/io/sentry/transport/HttpTransport.java:149]: okay, i will delete them.

regarding to my own testing 

this is how I initialize Sentry
image

MainPresenter sslcontext and hostnameverifier
image

image

envelope sent
image

**Comment by ninekaw9** [comment]
@bruno-garcia i pushed a commit which removed all bypassSecurity

**Comment by bruno-garcia** [comment]
Nice! Thanks!

For the record:

This is a breaking change, we need to document on the release note and update docs.
And it'll resolve #861

**Comment by marandaneto** [comment]
> @bruno-garcia i pushed a commit which removed all bypassSecurity

thanks for that, do you mind fixing the tests? CI is broken because the tests are not taking into account the new params on the ctor.

**Comment by ninekaw9** [comment]
> > @bruno-garcia i pushed a commit which removed all bypassSecurity
> 
> thanks for that, do you mind fixing the tests? CI is broken because the tests are not taking into account the new params on the ctor.

sorry, I am quite busy on my project right now. Plus, I don't have much experience in writing unit tests.

**Comment by marandaneto** [comment]
> > > @bruno-garcia i pushed a commit which removed all bypassSecurity
> > 
> > 
> > thanks for that, do you mind fixing the tests? CI is broken because the tests are not taking into account the new params on the ctor.
> 
> sorry, I am quite busy on my project right now. Plus, I don't have much experience in writing unit tests.

ok, i will take care of it, thanks

**Comment by marandaneto** [comment]
in favor of https://github.com/getsentry/sentry-java/pull/944

thanks @ninekaw9

**Pull request closed without merging**